ludwig/utils/strings_utils.py

```diff
@@ -243,6 +243,84 @@ class Vocabulary:
     unknown_symbol: str
     """Actual unknown symbol."""
 
+    prompt_template_num_tokens: int = 0
+    """The number of tokens in the prompt template.
+
+    If -1, then there is no prompt template.
+    """
+
+
+def _get_vocabulary(
+    tokenizer_type: str,
+    tokenizer,
+    vocab_file: str,
+    unknown_symbol: str,
+    add_special_symbols: bool,
+    padding_symbol: str,
+    unit_counts: Counter,
+    num_most_frequent: int,
+) -> Optional[List[str]]:
+    """Returns the vocabulary from the tokenizer_type, tokenizer, or vocab_file.
+
+    If there's no vocab_file or if the tokenizer has no set vocabulary (e.g. space_punct), then the vocabulary is
+    determined from the tokenized data.
+    """
```

Comment:
nit: Can we copy the docstring from the original `create_vocabulary` function into this helper function